x/wasm/keeper/msg_server.go

```diff
@@ -357,6 +358,83 @@ func (m msgServer) StoreAndInstantiateContract(goCtx context.Context, req *types
 	}, nil
 }
 
+// AddCodeUploadParamsAddresses adds addresses to code upload params
+func (m msgServer) AddCodeUploadParamsAddresses(goCtx context.Context, req *types.MsgAddCodeUploadParamsAddresses) (*types.MsgAddCodeUploadParamsAddressesResponse, error) {
+	if err := req.ValidateBasic(); err != nil {
+		return nil, err
+	}
+	authority := m.keeper.GetAuthority()
+	if authority != req.Authority {
+		return nil, errorsmod.Wrapf(types.ErrInvalid, "invalid authority; expected %s, got %s", authority, req.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	params := m.keeper.GetParams(ctx)
+	if params.CodeUploadAccess.Permission != types.AccessTypeAnyOfAddresses {
+		return nil, errorsmod.Wrap(types.ErrInvalid, "permission")
+	}
+
+	addresses := params.CodeUploadAccess.Addresses
+	size := len(addresses)
+	for _, newAddr := range req.Addresses {
+		for i := 0; i < size; i++ {
+			if addresses[i] == newAddr {
+				return nil, errorsmod.Wrap(types.ErrInvalid, fmt.Sprintf("address already exists %s", newAddr))
```

Comment:
Let's silently skip the address when it exists already. This makes it more resilient